documentation/design/proposals/authn_authz_for_1.0.adoc

```diff
@@ -0,0 +1,329 @@
+
+To support the simple getting started use case, if there are no `AccessControlService` defined in the infra namespace,
+the system will behaves as if there is an `AccessControlService` with the anonymous identity provider defined.  For
+authorization, it will be as if an authorisation policy grant all access to all addresses exists.
+
+As soon as an `AccessControlService` is defined in the infra namespace, the default described here becomes unavailable for
+use.
+
+===== Go AMQP server implementation
+
+The Access Control Server's Go AMQP server requirements are quite limited.   It does not require a full-blown AMQP server
+implementation.  It is sufficient to handle the AMQP SASL performatives and the AMQP Open/Close performatives.  This can
+be implemented within the Access Control Server code-base.  It will borrow codec implementation from Azure AMQP client.
+
+==== EnMasse Operator
+
+The EnMasse Operator will act as the controller for `AccessControlService`, `MessagingUser`, `MessagingRole`, and `MessagingRoleBinding` resources and be responsible for maintaintence of the status sections.  For `MessagingUser`
+resources it will be responsible for salting passwords. 
+
+The EnMasse operator will watch for `MessagingTenants` resources.  For each it will and define a Kubernetes service
+exposing the AMQP SASL endpoint of the Access Control Servcie  The MessagingInfra controller will configure a router listener/authServicePlugin _per tenant_  with a reference to the correct  AMQP SASL endpoint.
+
+==== Messaging Client
+
+When the messaging clients authenticate, in some cases the username must be qualified with an `@DOMAIN` suffix.  The qualification must be used when authenticatiing with an identity from an external LDAP or external AMQP SASL server.
+
+For theinternal identity provider, the domain suffix may be omitted.  The service accounts, the username is always `@@serviceaccount@@`.  For client auth, the identity is the common name from the certificate.
+
+==== Qpid Dispatch Router
+
+This proposal reuses the existing `authServicePlugin` of the Qpid Dispatch Router unchnaged.
+
+=== Resources
+
+The new API will consist of the following types at the infrastructure level:
+
+* `AccessControlService`: used to define an instance of an accesss control service. The `AccessControlService` will define
+the parameters to establish connection with external identity provider (if any).  Selectors will be to configure to which namespaces it should be applied.
+
+* Configuration provided by `MessagingInfra` will define the Access Control Server deployment.
+
+At the tenant level:
+
+* `MessagingTennant`: can be used to specify a specific `AccessControlService` to be used for the tenant.
+
+* `MessagingUser` (v2beta1): used with the internal identity source to provide a messaging username/password pairing.  The password will be stored in the resource a secure fashion.  `MessagingUser` is scoped to the messaging tenant. It cannot
```

Comment:
Should we use v1beta2 for all the 1.0 resources, with the goal of moving them to v1 once we are happy with it and ready to release?